util/annotations/annotations.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package annotations
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Annotations is a general wrapper for text that is shown in the
+// Prometheus/Grafana UI along with the query results.
+type Annotations map[string]error
+
+func (a *Annotations) AddAnnotation(err error) {
+	(*a)[err.Error()] = err
+}
+
+func InitAnnotations(err error) Annotations {
+	a := make(Annotations)
+	a.AddAnnotation(err)
+	return a
+}
+
+func (a *Annotations) Merge(aa Annotations) {
+	if *a == nil {
+		*a = make(Annotations)
+	}
+	for key, val := range aa {
+		(*a)[key] = val
+	}
+}
+
+func InitAnnotationsAndMerge(err error, aa Annotations) Annotations {
+	a := InitAnnotations(err)
+	a.Merge(aa)
+	return a
+}
+
+func (a *Annotations) AsErrArray() []error {
+	arr := make([]error, 0, len(*a))
+	for _, err := range *a {
+		arr = append(arr, err)
+	}
+	return arr
+}
+
+func (a *Annotations) AsStrArray() []string {
+	arr := make([]string, 0, len(*a))
+	for _, err := range *a {
+		arr = append(arr, err.Error())
```

Comment:
The keys in the maps are already the strings created by `err.Error()`. You can use them directly.